app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -60,6 +62,16 @@ class SettingsFragment : PreferenceFragmentCompat(), CoroutineScope, AccountObse
         super.onCreate(savedInstanceState)
         job = Job()
         updateSignInVisibility()
+
+        preferenceManager.sharedPreferences.registerOnSharedPreferenceChangeListener { sharedPreferences, key ->
+            try {
+                context!!.components.analytics.metrics.track(Event.PreferenceToggled
```

Comment:
I think this is safe because onCreate comes right after onAttach, but I don't think we should crash if context is null here.